graftm/tree_decorator.py

```diff
@@ -1,309 +1,187 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                #
+# GNU General Public License for more details.                                #
+#                                                                             #
+# You should have received a copy of the GNU General Public License           #
+# along with this program. If not, see <http://www.gnu.org/licenses/>.        #
+#                                                                             #
+###############################################################################
+__author__ = "Joel Boyd, Ben Woodcroft"
+__copyright__ = "Copyright 2015"
+__credits__ = ["Joel Boyd"]
+__license__ = "GPL3"
+__version__ = "0.0.1"
+__maintainer__ = "Joel Boyd"
+__email__ = "joel.boyd near uq.net.au"
+__status__ = "Development"
+###############################################################################
 # System imports
-from skbio import TreeNode
-import argparse
 import logging
+import tempfile
+import subprocess
+import numpy as np
 import os
-import sys
+import random
+from scipy import stats
+from itertools import combinations
+from collections import Counter
+from skbio.tree import TreeNode
+from graftm.taxonomy import Taxonomy
 
+# Local imports
+################################################################################
 ################################################################################
-################################ - Statics - ###################################
-MAX_RESOLUTION = 7
-
 ################################################################################
-################################## - Code - ####################################
 
 class TreeUnrootedException(Exception): 
     pass
 
-class TreeDecorator:
-    '''Yet another class that decorates trees, only the way I want it.'''
-    
-    
-    def __init__(self, 
-                 tree,
-                 taxonomy,
-                 seqinfo=None):
+#-------------------------------------------------------------------------------
+
+class MalformedTreeException(Exception):
+    pass
+
+#-------------------------------------------------------------------------------
+
+class TreeDecorator: # Room to grow!
+    '''
+    Decorate trees with information!
+    '''
+    GREENGENES_PREFIXES = set(['k__',
+                               'p__',
+                               'c__',
+                               'o__',
+                               'f__',
+                               'g__',
+                               's__'])
+
+    def __init__(self, tree, no_unique_names):
         '''
-        Set up empty lists and open and import tree object, and load the
-        taxonomy into a hash
+        Parameters
+        ----------
+        tree: skbio TreeNode obj
+            http://scikit-bio.org/docs/latest/generated/skbio.tree.TreeNode.html#skbio.tree.TreeNode
+        no_unique_names : bool
+            True of False, whether names will be made non-redundant if 
```

Comment:
I think it is more understandable to avoid double negatives, so better to use `generate_unique_names` so to make unique names.

But more generally, I think this style of class where there is a bunch of methods to setup the parameters for a single call (here `taxonomy_partition`) isn't optimal because there are several ways to provide parameters - via the instance variable, in the init and as parameters directly to taxonomy_partition. THere's also 2 types of methods, those for before decoration and those for after.

I reckon a better setup would be to Have a taxonomy decorator class, and then a separated DecoratedTree class, which has the write_taxonomy methods and etc. 

Make sense?